src/validator/validator.test.ts

```diff
@@ -730,6 +730,63 @@ describe('Nested objects', () => {
       expect(results[0].message).toBe(`${messages[i]}`)
     }
   })
+
+  it('Should threat `v.json()` as a nested array type only inside an array', async () => {
+    const json = {
+      blog: {
```

Comment:
If we add the `author.name` such as below:

```ts
      blog: {
        totalPosts: '3',
        posts: [
          {
            id: 123,
            title: 'JavaScript',
            author: {
              name: 'foo',
            },
          },
          {
            id: '456',
            title: 'Framework',
            author: {
              name: 'bar',
            },
          },
          {
            id: 789,
            title: 'Hono',
            author: {
              name: 'baz',
            },
          },
        ],
        highlightedPostId: '123',
      },
```

and write the validator below:

```ts
    const v = new Validator()
    const vObject = v.object('blog', (v) => ({
      posts: v.array('posts', (v) => ({
        author: v.object('author', (v) => ({
          name: v.json('name'),
        })),
      })),
    }))
```

then, we expect it to be valid, but it will no be valid. So, I think we have to remove this line:

https://github.com/honojs/hono/blob/8ed7763c38e153ee78860724582ffd0f0d96a845/src/validator/validator.ts#L111

Reply:
good point, I guess it was added there to reset the `isArray` field? that shouldn't be necessary anymore right?

Reply:
Yes, this line is not needed.

Reply:
The tests you write look good!